AM_C_PROTOTYPES was only useful for very ancient C compilers that were
not ANSI C compliant. It has been removed from automake. This commit
shows the removal and details about the macro:

http://git.savannah.gnu.org/cgit/automake.git/commit/?id=0cbcf393dbdaa2ff36b11cf2516675e78eada49f


Index: popt-1.16/configure.ac
===================================================================
--- popt-1.16.orig/configure.ac	2014-08-24 14:36:59.142006790 +0100
+++ popt-1.16/configure.ac	2014-08-24 14:38:40.015988208 +0100
@@ -46,7 +46,6 @@
 AC_SYS_LARGEFILE
 
 AC_ISC_POSIX
-AM_C_PROTOTYPES
 
 AC_CHECK_HEADERS(float.h fnmatch.h glob.h langinfo.h libintl.h mcheck.h unistd.h)
 
